src/DisplayPlug.vala

```diff
@@ -46,13 +49,38 @@ public class Display.Plug : Switchboard.Plug {
             displays_view = new DisplaysView ();
             var action_bar = new Gtk.ActionBar ();
             action_bar.get_style_context ().add_class (Gtk.STYLE_CLASS_INLINE_TOOLBAR);
+            
+            detect_touchscreen ();
+
+            var schema_source = GLib.SettingsSchemaSource.get_default ();
```

Comment:
This could've been optimized by checking if the system does actually have a touchscreen. There is no need to check if there is a touchscreen schema when there is no touchscreen device present.